tests/test_staking_service.py

```diff
@@ -0,0 +1,657 @@
+            return False
+
+def get_service_token(test_instance) -> Optional[int]:
+    """Get service token from runtime config."""
+    try:
+        # Find the service config directory
+        services_dir = Path(os.getcwd()) / ".operate" / "services"
+        if not services_dir.exists():
+            test_instance.logger.error(f"Services directory not found at {services_dir}")
+            return None
+            
+        # Find service directory by looking for config.json
+        for service_dir in services_dir.iterdir():
+            config_file = service_dir / "config.json"
+            if config_file.exists():
+                with open(config_file) as f:
+                    runtime_config = json.load(f)
+                
+                # Get token from chain_data
+                chain_name = runtime_config.get("home_chain", "gnosis")
+                chain_config = runtime_config.get("chain_configs", {}).get(chain_name, {})
+                chain_data = chain_config.get("chain_data", {})
+                token = chain_data.get("token")
+                
+                if token:
+                    test_instance.logger.info(f"Found service token: {token}")
+                    return token
+                    
+        test_instance.logger.error("No service token found in runtime config")
+        return None
+        
+    except Exception as e:
+        test_instance.logger.error(f"Error getting service token: {e}")
+        return None
+
+def verify_staking(test_instance) -> bool:
+    """Verify staking status for a service."""
+    checker = StakingStatusChecker(test_instance.config_path, test_instance.logger)
+    
+    try:
+        # Find and load runtime config
+        services_dir = Path(os.getcwd()) / ".operate" / "services"
+        runtime_config = None
+        
+        for service_dir in services_dir.iterdir():
+            config_file = service_dir / "config.json"
+            if config_file.exists():
+                with open(config_file) as f:
+                    runtime_config = json.load(f)
+                break
+                
+        if not runtime_config:
+            test_instance.logger.error("No runtime config found")
+            return False
+        
+        # Get chain data from runtime config
+        chain_name = runtime_config.get("home_chain", "gnosis")
+        chain_data = runtime_config.get("chain_configs", {}).get(chain_name, {}).get("chain_data", {})
+        
+        # Get service token and staking program
+        service_token = chain_data.get("token")
+        staking_program_id = chain_data.get("user_params", {}).get("staking_program_id")
+        
+        if not service_token:
+            test_instance.logger.error("Could not find service token")
+            return False
+            
+        if not staking_program_id or staking_program_id == "no_staking":
+            test_instance.logger.info("Service is not using staking")
+            return True
```

Comment:
probably pass the expected staking program in the function parameter, and match that here